### PR TITLE
Upgrade Node to 22 and update the dependenies to be compatible with node 22 as node v16 is depricated 

### LINKED ISSUE #553: Upgrade Node to version 22
### Description:

Node 16 is deprecated, it is receiving no sec updates, so it would be better to update node to 22, as it will soon move to LTS, and updating to any higher versions(20, as 19 is also deprecated) will need `next` to be updated to 14(Will be a huge refactor). So updating node to 22 would just do fine as it won't require nextjs update

# Scope

1. Upgrade node to v22.x in both cli/ and webserver/
2. See if any dependency requires an update
3. Update the docker setup to use node 22.x instead of 16
4. Do the necessary clean-up if node16 still exists in repo

### PR DESCRIPTION
**Pull request by thesujai**
## Linked Issue(s)

Fixes #553 

## Acceptance Criteria fulfillment

- [x] Upgrade and pin node to 22.x in both cli and webser
- [x] Upgrade the necessary dependency and run yarn install to ensure proper updates to yarn.lock file
- [x] Clean up node 16 from the repo

## Proposed changes (including videos or screenshots)

NA

## Further comments

NA

### CODE REVIEW

File: Dockerfile.dev [hunk 1]
```diff
@@ -44,7 +44,9 @@ RUN mv /app/database-docker/db/ /app/ && rm -rf /app/database-docker/
 RUN echo "host all  all    0.0.0.0/0  md5" >> /etc/postgresql/15/main/pg_hba.conf
 RUN echo "listen_addresses='*'" >> /etc/postgresql/15/main/postgresql.conf
 RUN sed -i "s/^port = .*/port = ${DB_PORT}/" /etc/postgresql/15/main/postgresql.conf
-RUN npm install --global yarn --force
+RUN curl -fsSL https://deb.nodesource.com/setup_18.x | bash - \
```
Comment (thesujai) [inline Dockerfile.dev:47]: I am not sure whether this was required or not. As the base image in this dockerfile is `middlewareeng/oss-base`, and I suppose this image is being built from Dockerfile.prebuilt(Where I upgraded to node 18)
If my understanding of the code is correct then changes to this file can be removed and a new commit can be made to `middlewareeng/oss-base` with the updates in the Dockerfile.prebuilt

I am not so sure about my understanding of this

File: Dockerfile.dev [hunk 2]
```diff
@@ -44,7 +44,9 @@ RUN mv /app/database-docker/db/ /app/ && rm -rf /app/database-docker/
 RUN echo "host all  all    0.0.0.0/0  md5" >> /etc/postgresql/15/main/pg_hba.conf
 RUN echo "listen_addresses='*'" >> /etc/postgresql/15/main/postgresql.conf
 RUN sed -i "s/^port = .*/port = ${DB_PORT}/" /etc/postgresql/15/main/postgresql.conf
-RUN npm install --global yarn --force
+RUN curl -fsSL https://deb.nodesource.com/setup_22.x | bash - \
+    && apt-get install -y nodejs \
+    && npm install --global yarn
```
Comment (adnanhashmi09) [inline Dockerfile.dev:49]: Why is this required in the dev setup, if we are already using the pre-built base image. This would slow down the dev process.

**Review by adnanhashmi09 — CHANGES_REQUESTED** [review]

File: Dockerfile.dev (hunk 1 above)
Comment (adnanhashmi09) [reply Dockerfile.dev]: > I am not sure whether this was required or not. As the base image in this dockerfile is `middlewareeng/oss-base`, and I suppose this image is being built from Dockerfile.prebuilt(Where I upgraded to node 18) If my understanding of the code is correct then changes to this file can be removed and a new commit can be made to `middlewareeng/oss-base` with the updates in the Dockerfile.prebuilt
> 
> I am not so sure about my understanding of this

yep. Not needed here

File: Dockerfile.dev (hunk 2 above)
Comment (thesujai) [reply Dockerfile.dev:49]: Can you push the pre-built base image with the latest changes so I can remove this one? As the old base image has node 16

Comment (thesujai) [reply Dockerfile.dev:49]: Push into dockerhub I mean

**Comment by adnanhashmi09** [comment]
I will update the base image today @thesujai.

**Comment by thesujai** [comment]
Reverted it! @adnanhashmi09

**Comment by adnanhashmi09** [comment]
I have pushed the latest build with node 22 of oss-base. Please test your PR once and check if everything works. Please rebase and test your pr once.

**Review by adnanhashmi09 — APPROVED** [review]
